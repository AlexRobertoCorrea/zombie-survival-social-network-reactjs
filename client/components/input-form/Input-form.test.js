import React from 'react';

import InputForm from './Input-form';

describe('When selectOptions is passed at props', () => {
  const field = {
    label: '',
    id: '',
    rules: '',
    errorMessage: '',
    helperText: '',
    required: false,
    type: '',
  };
  const maxLength = 50;
  const selectOptions = [{
    label: '',
    value: ''
  }];
  const form = {
    $: () => ({
      ...field
    }),
    errors: () => ({ [field.id]: '' })
  };
  
  const inputForm = shallow(<InputForm
    form={form}
    field={field}
    maxLength={maxLength}
    selectOptions={selectOptions}
  />);
  
  it('renders the inputForm as expected', () => {
    expect(inputForm).toMatchSnapshot();
  });
});

describe('When selectOptions is not passed at props', () => {
  const field = {
    label: '',
    id: '',
    rules: '',
    errorMessage: '',
    helperText: '',
    required: false,
    type: '',
  };
  const maxLength = 50;
  const form = {
    $: () => ({
      ...field
    }),
    errors: () => ({ [field.id]: '' })
  };
  
  const inputForm = shallow(<InputForm
    form={form}
    field={field}
    maxLength={maxLength}
  />);
  
  it('renders the inputForm as expected', () => {
    expect(inputForm).toMatchSnapshot();
  });
});
