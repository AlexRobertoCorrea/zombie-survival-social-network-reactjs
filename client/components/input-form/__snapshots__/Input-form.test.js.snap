// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When selectOptions is not passed at props renders the inputForm as expected 1`] = `
<WithStyles(FormControl)
  error={false}
  fullWidth={true}
  key=""
  margin="dense"
  required={false}
>
  <WithStyles(InputLabel) />
  <WithStyles(Input)
    id="margin-normal"
    inputProps={
      Object {
        "maxLength": 50,
      }
    }
    type=""
  />
  <div />
</WithStyles(FormControl)>
`;

exports[`When selectOptions is passed at props renders the inputForm as expected 1`] = `
<WithStyles(FormControl)
  error={false}
  fullWidth={true}
  key=""
  margin="dense"
  required={false}
>
  <WithStyles(InputLabel) />
  <WithStyles(Select)
    inputProps={
      Object {
        "id": "",
        "name": "",
      }
    }
    onChange={[Function]}
  >
    <WithStyles(MenuItem)
      key=""
      value=""
    />
  </WithStyles(Select)>
  <WithStyles(FormHelperText) />
  <div />
</WithStyles(FormControl)>
`;
